libs/forklift_inventory.py

```
@@ -406,10 +406,106 @@ def __init__(self, client: DynamicClient, provider_name: str, namespace: str) ->
+            for sample in TimeoutSampler(
+                wait_timeout=timeout,
+                sleep=sleep,
+                func=lambda: self.hosts,
+            ):
+                if sample:
+                    LOGGER.info(f"Found {len(sample)} hosts in inventory for provider '{self.provider_name}'")
+                    return sample
+        except TimeoutExpiredError:
+            raise TimeoutExpiredError(
+                f"No hosts appeared in Forklift inventory for provider '{self.provider_name}' after {timeout}s"
+            )
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)